modules/new_profile/php/profile.class.inc

```diff
@@ -120,8 +135,126 @@ class Profile extends \NDB_Page
                 ->withStatus(405)
                 ->withHeader("Allow", "GET,POST");
         }
+    }
 
+    /**
+     * Processes the values and saves to database
+     *
+     * @param array $values the user data received..
+     *
+     * @return array
+     */
+    function processValidity($values)
+    {
+        // Validate (dateTaken, dateTakenConfirm, sex) not set.
+        if (!isset($values['dateTaken'])
+            || !isset($values['dateTakenConfirm'])
+            || !isset($values['sex'])
+        ) {
+            return array('error' => 'Incomplete candidate fields.');
+        }
+        // Validate dob.
+        if (strlen($values['dateTaken']) > 2
+            && !$this->_checkDate($values['dateTakenConfirm'])
+        ) {
+            return array('error' => 'Birth date is not a valid date.');
+        } else if ($values['dateTaken'] != $values['dateTakenConfirm']) {
+            // Validate (dob1 & dob2) are not equal.
+            return array('error' => 'Birth date confirmation mismatch.');
+        }
+        $config =& \NDB_Config::singleton();
+        // Validate edc.
+        if ($config->getSetting('useEDC') == 'true'
+            && is_array($values['edcDateTaken'])
+            && is_array($values['edcDateTakenConfirm'])
+        ) {
+            if (strlen(implode($values['edcDateTaken'])) > 2
+                && !$this->_checkDate($values['edcDateTaken'])
+            ) {
+                return array('error' => 'EDC is not a valid date.');
+            } else if ($values['edcDateTaken'] != $values['edcDateTakenConfirm']) {
+                return array('error' => 'Estimated Due date fields must match.');
+            }
+        } else {
+            $values['edcDateTaken']        = null;
+            $values['edcDateTakenConfirm'] = null;
+        }
+        // Validate Gender.
+        if (strcasecmp($values['sex'], 'Male') !== 0
+            && strcasecmp($values['sex'], 'Female') !== 0
+            && strcasecmp($values['sex'], 'Other') !== 0
+            && strcasecmp($values['sex'], 'Unknown') !== 0
+        ) {
+            return array('error' => 'Invalid sex option.');
+        }
+        // Validate PSCID.
+        $pscidSettings = $config->getSetting('PSCID');
+        $user          =& \User::singleton();
+        if ($pscidSettings['generation'] == 'user') {
+            // Setup Database object.
+            $db = \Database::singleton();
+            if (empty($values['site'])) { // user is in only one site.
+                $centerIDs = $user->getData('CenterIDs');
+                $centerID  = $centerIDs[0];
+                $site      =& \Site::singleton($centerID);
+            } else {
+                // user has multiple sites, so validate
+                // PSCID against the Site selected.
+                $site =& \Site::singleton($values['site']);
+            }
+            if (empty($values['PSCID'])) {
+                return array('error' => 'PSCID must be specified.');
```

Comment:
on testing, i keep getting the error 'PSCID must be specified.' even though `formData.pscid` is not empty. the value must not be appending to the request some how

<img width="754" alt="Screenshot 2019-05-28 19 48 08" src="https://user-images.githubusercontent.com/17415878/58519463-b0e3e380-8181-11e9-856c-ae3b14f24e4a.png">

Reply:
Hey @zaliqarosli, can you test again?
I think the new commit should fix it. The backend logic was using all caps for PSCID and when it should've been lowercase to reflect the data from the frontend form.